pkg/controller/migmigration/backup.go

```diff
@@ -401,6 +402,222 @@ func (t *Task) deleteBackups() error {
 	return nil
 }
 
+// Delete stale Velero Backups/Restores, set condition indicating
+// Velero restart should be performed later if anything was deleted.
+func (t *Task) deleteStaleVeleroCRs() error {
+	totalDeleted := 0
+	nBackupsDeleted, err := t.deleteStaleBackups()
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	totalDeleted += nBackupsDeleted
+	nRestoresDeleted, err := t.deleteStaleRestores()
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	totalDeleted += nRestoresDeleted
+
+	if totalDeleted > 0 {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     StaleVeleroCRsDeleted,
+			Status:   True,
+			Category: migapi.Required,
+			Message: fmt.Sprintf("Deleted stale Velero CRs: %v Backups,  %v Restores. "+
+				"Velero will be restarted.", nBackupsDeleted, nRestoresDeleted),
+			Durable: true,
+		})
+	}
+	return nil
+}
+
+// Delete stale Restic CRs, set condition indicating
+// Restic Pod restarts may be needed, but we leave this up to the user.
+func (t *Task) deleteStaleResticCRs() error {
+	totalDeleted := 0
+	pvbsDeleted, err := t.deleteStalePVBs()
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	totalDeleted += pvbsDeleted
+	pvrsDeleted, err := t.deleteStalePVRs()
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	totalDeleted += pvrsDeleted
+
+	if totalDeleted > 0 {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     StaleResticCRsDeleted,
+			Status:   True,
+			Category: migapi.Required,
+			Message: fmt.Sprintf("Deleted stale Restic CRs: "+
+				"%v PodVolumeBackups,  %v PodVolumeRestores. "+
+				"Manual Restic Pod restart may be required if Restic is stuck.", pvbsDeleted, pvrsDeleted),
+			Durable: true,
+		})
+	}
+	return nil
+}
+
+func (t *Task) migrationUIDisRunning(migrationUID string) (bool, error) {
+	corrKey, _ := t.Owner.GetCorrelationLabel()
+	corrLabel := map[string]string{corrKey: migrationUID}
+	migrationList := migapi.MigMigrationList{}
+	err := t.Client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(corrLabel),
+		&migrationList,
+	)
+	if err != nil {
+		return false, liberr.Wrap(err)
+	}
+	migrationIsRunning := false
+	for _, migration := range migrationList.Items {
+		if migration.Status.Phase != Completed {
+			migrationIsRunning = true
+		}
+	}
+	return migrationIsRunning, nil
+}
+
+// Delete stale Velero Backups in the controller namespace to empty
+// the work queue for next migration.
+func (t *Task) deleteStaleBackups() (int, error) {
+	nDeleted := 0
+	srcClient, err := t.getSourceClient()
```

Comment:
We should probably consider both src and dest here, since a stale backup on the dest cluster will still block restores (and vice versa)

Reply:
Changed to only restart Velero on the cluster where Backups/Restores were deleted.

Reply:
I missed the meaning of your comment on my first read. I need to make sure I also:

1. delete stale backups from the destination cluster
2. delete stale restores from the src cluster